hadoop-yarn-project/hadoop-yarn/hadoop-yarn-client/src/test/java/org/apache/hadoop/yarn/client/cli/TestRouterCLI.java

```diff
@@ -120,7 +122,10 @@ public void testDeregisterSubCluster() throws Exception {
     ByteArrayOutputStream dataOut = new ByteArrayOutputStream();
     System.setOut(new PrintStream(dataOut));
     oldOutPrintStream.println(dataOut);
-    String[] args = {"-deregisterSubCluster", "-c", "SC-1"};
+    String[] args = {"-deregisterSubCluster", "-sc", "SC-1"};
```

Comment:
Aren't we changing the interface here?
I don't think we should break backwards compatibility.

Reply:
Wait, where does the -c came from?

Reply:
Apologies for the oversight. In the command, we wrote `-sc` as the option, but the actual input requires `-c`.

RouterCLI#ADMIN_USAGE
```
protected final static Map<String, UsageInfo> ADMIN_USAGE =
      ImmutableMap.<String, UsageInfo>builder().put("-deregisterSubCluster",
        new UsageInfo("[-sc|subClusterId [subCluster id]]",
        "deregister subCluster, if the interval between the heartbeat time of the subCluster " 
```

RouterCLI#handleDeregisterSubCluster
```
Options opts = new Options();
    opts.addOption("deregisterSubCluster", false,
        "Refresh the hosts information at the ResourceManager.");
    Option gracefulOpt = new Option("c", "clusterId", true,
        "Wait for timeout before marking the NodeManager as decommissioned.");
    gracefulOpt.setOptionalArg(true);
    opts.addOption(gracefulOpt);
```

In this pr, I have fixed this issue by allowing the use of `-sc` or `--subClusterId` to specify the subCluster.

Cmd1 : `yarn routeradmin -deregisterSubCluster`

<img width="1089" alt="image" src="https://github.com/apache/hadoop/assets/55643692/e7fcf521-0ca3-4717-8110-1ad8f20916cd">

Cmd2 : `yarn routeradmin -deregisterSubCluster -sc ClusterTest-Yarn1`

<img width="1078" alt="image" src="https://github.com/apache/hadoop/assets/55643692/5993e329-c41e-4378-b1d4-8d751fff6140">

Cmd3 :`yarn routeradmin -deregisterSubCluster --subClusterId ClusterTest-Yarn1`

<img width="1109" alt="image" src="https://github.com/apache/hadoop/assets/55643692/43824c44-b0fb-4a97-a1c6-a83804ec9961">

Reply:
It is weird to have a single dash for two letters.
It should be `--sc`, no?

Reply:
Thanks for your question! You are correct that in general, a single character follows the hyphen "-".  Our short option hopes to express `subcluster`, maybe `sc` is better than `c`?

- RouterCLI#handleDeregisterSubCluster
```
private static final String OPTION_SC = "sc";
private static final String OPTION_SUBCLUSTERID = "subClusterId";

Option subClusterOpt = new Option(OPTION_SC, OPTION_SUBCLUSTERID, true,
        "The subCluster can be specified using either the '-sc' or '--subCluster' option. " +
        " If the subCluster's Heartbeat Timeout, it will be marked as 'SC_LOST'.");
```

- org.apache.commons.cli.Option

For Options, we set short options using a hyphen `-` and long options using two hyphens `--` when using the command.

```
 /**
    * Creates an Option using the specified parameters.
    *
    * @param option short representation of the option
    * @param longOption the long representation of the option
    * @param hasArg specifies whether the Option takes an argument or not
    * @param description describes the function of the option
    *
    * @throws IllegalArgumentException if there are any non valid Option characters in {@code opt}.
    */
    public Option(final String option, final String longOption, final boolean hasArg, final String description) throws IllegalArgumentException {
        // ensure that the option is valid
        this.option = OptionValidator.validate(option);
        this.longOption = longOption;

        // if hasArg is set then the number of arguments is 1
        if (hasArg) {
            this.argCount = 1;
        }

        this.description = description;
    }
```

Our short option is set as `sc`, so it should be used as `-sc` when using the command. The long option is set as `subClusterId`, so it should be used as `--subClusterId` when using the command.